src/management-system-v2/components/auth-can.tsx

```diff
@@ -69,8 +70,12 @@ export const AuthCan: FC<PropsWithChildren<AuthCanProps>> = ({
   return fallback || null;
 };
 
-export const useEnvironment = () => {
-  const pathname = usePathname();
-  const environmentId = pathname.split('/')[1];
-  return decodeURIComponent(environmentId);
+export const useEnvironment = (returnUserId: boolean = false) => {
```

Comment:
The reason was that callers of this hook have one of two interests: Either they just want the `spaceId` and don't care if it is a personal space or not.
Or they care if it is a personal space (like `spaceURL()`) and only care about the id if it is not.
If this hook only ever returned the userId, the caller would have to perform another check to find out if it is a personal space or not. I now changed this hook to return an object with both information to make the use easier.

Reply:
I see, I like the solution. However, I don't know if I quite agree with the getCurrentEnvironment function returning whether the environment is an organization or not, as (at least conceptually) I had planned that most of the code works the same for personal spaces as for organization, meaning that there wouldn't be the need to differentiate between them. And that the differentiation would be done "internally" with abilities.

Reply:
Yeah, I agree that we shouldn't use that to differentiate between features. But for things like links, redirects and reconstructing paths it might be useful, since those aren't really abilities.